packages/expo-av/src/ExponentAV.web.ts

```diff
@@ -39,80 +40,118 @@ function getStatusFromMedia(media?: HTMLMediaElement): PlaybackStatus {
   ): Promise<PlaybackStatus> {
-    return await this.getStatusForVideo(element);
+    return getStatusFromMedia(element);
   },
   async unloadForVideo(element: HTMLMediaElement): Promise<PlaybackStatus> {
-    return await this.getStatusForVideo(element);
+    return getStatusFromMedia(element);
   },
   async setStatusForVideo(
     element: HTMLMediaElement,
     status: PlaybackStatusToSet
   ): Promise<PlaybackStatus> {
-    if (status.positionMillis !== undefined) {
-      element.currentTime = status.positionMillis / 1000;
-    }
-    // if (status.progressUpdateIntervalMillis !== undefined) {
-    //   element.progressUpdateIntervalMillis = status.progressUpdateIntervalMillis;
-    // }
-    // if (status.seekMillisToleranceBefore !== undefined) {
-    //   element.seekMillisToleranceBefore = status.seekMillisToleranceBefore;
-    // }
-    // if (status.seekMillisToleranceAfter !== undefined) {
-    //   element.seekMillisToleranceAfter = status.seekMillisToleranceAfter;
-    // }
-    // if (status.shouldCorrectPitch !== undefined) {
-    //   element.shouldCorrectPitch = status.shouldCorrectPitch;
-    // }
-    if (status.shouldPlay !== undefined) {
-      if (status.shouldPlay) {
-        element.play();
-      } else {
-        element.pause();
-      }
-    }
-    if (status.rate !== undefined) {
-      element.playbackRate = status.rate;
-    }
-    if (status.volume !== undefined) {
-      element.volume = status.volume;
-    }
-    if (status.isMuted !== undefined) {
-      element.muted = status.isMuted;
-    }
-    if (status.isLooping !== undefined) {
-      element.loop = status.isLooping;
-    }
-    return await this.getStatusForVideo(element);
+    return setStatusForMedia(element, status);
   },
   async replayVideo(
     element: HTMLMediaElement,
     status: PlaybackStatusToSet
   ): Promise<PlaybackStatus> {
-    return await this.setStatusForVideo(element, status);
+    return setStatusForMedia(element, status);
   },
   /* Audio */
   async setAudioMode() {},
   async setAudioIsEnabled() {},
+  async getStatusForSound(element: HTMLMediaElement) {
+    return getStatusFromMedia(element);
+  },
+  async loadForSound(
+    nativeSource: string | { uri: string; [key: string]: any },
+    fullInitialStatus: PlaybackStatusToSet
+  ): Promise<[HTMLMediaElement, PlaybackStatus]> {
+    const source = typeof nativeSource === 'string' ? nativeSource : nativeSource.uri;
+    const media = new Audio(source);
+
+    media.ontimeupdate = () => {
+      SyntheticPlatformEmitter.emit('didUpdatePlaybackStatus', {
+        key: media,
+        status: getStatusFromMedia(media),
+      });
+    };
 
-  async getStatusForSound() {},
-  async setErrorCallbackForSound() {},
-  async loadForSound() {},
-  async unloadForSound() {},
-  async setStatusForSound() {},
-  async replaySound() {},
+    media.onerror = (event: Event | string, source?: string, fileno?: number, columnNumber?: number, error?: Error) => {
```

Comment:
The properties event, source, fileno, etc... are properties of ErrorEvent objects -> not sure these get passed in as separate arguments. Also for media elements (video, audio), I believe `media.error` gets set when there's an `error` event, and we could reliably get a JS error object from there:

```js
media.onerror = () => {
  // media.error.message and media.error.code are defined
  // https://developer.mozilla.org/en-US/docs/Web/API/MediaError
};
```